pkg/controller/operators/catalog/subscription/syncer.go

```diff
@@ -66,10 +66,19 @@ func (s *subscriptionSyncer) Sync(ctx context.Context, event kubestate.ResourceE
 	switch event.Type() {
 	case kubestate.ResourceAdded:
 		initial = initial.Add()
+		metrics.EmitSubMetric(res)
 	case kubestate.ResourceUpdated:
+		oldSubscription, err := s.client.OperatorsV1alpha1().Subscriptions(res.GetNamespace()).Get(context.TODO(), res.GetName(), metav1.GetOptions{})
```

Comment:
@njhale could you explain what you mean by "from the cache"? The only subscription I'm aware of is the one on cluster, which I decided to grab and delete the metrics for before updating the on cluster subscription.

Reply:
> could you explain what you mean by "from the cache"? 

I mean the one being passed from the informer to this `Sync` function (via `event.Resource()`.

> The only subscription I'm aware of is the one on cluster, which I decided to grab and delete the metrics for before updating the on cluster subscription.

So here you are being given `res.Resource()` which _hopefully_ represents the latest `Subscription` and you are comparing it against the `Subscription` returned from the client, which you _know_ is the latest. Since `res.Resource()` is coming from an informer -- a watch -- chances are pretty good that it's up to date, which means you'll only ever trigger the nested if-statement below in the rare case the `Subscription` is updated between when the event is picked up by the informer and when this sync is called.